### PR TITLE
docs: improve dotnet getting started

### PR DESCRIPTION
**Pull request by debs-obrien**
full re-write of dotnet guide

### CODE REVIEW

File: docs/src/running-tests-csharp.md [hunk 1]
```diff
@@ -7,57 +7,132 @@ title: "Running and debugging tests"
...
 
-  ```bash
-  dotnet test -- Playwright.BrowserName=webkit
-  ```
+```bash
+dotnet test -- Playwright.BrowserName=webkit
+```
 
-- Running Tests on multiple browsers
+To specify multiple browsers to run your tests on, use the `--project` flag multiple times followed by the name of each browser.
```
Comment (mxschmitt) [inline docs/src/running-tests-csharp.md:69]:
```suggestion
```

Thats not a thing in .NET.

**Review by mxschmitt — CHANGES_REQUESTED** [review]

File: docs/src/running-tests-csharp.md (hunk 1 above)
Comment (debs-obrien) [reply docs/src/running-tests-csharp.md:69]: ooops, get left over from js

File: docs/src/trace-viewer-intro-csharp-java-python.md [hunk 2]
```diff
@@ -173,19 +172,19 @@ public class UnitTest1 : PageTest
     public async Task TestCleanup()
     {
         // This will produce e.g.:
-        // bin/Debug/net8.0/playwright-traces/PlaywrightTests.UnitTest1.zip
+        // bin/Debug/net8.0/playwright-traces/PlaywrightTests.ExampleTest.GetStartedLink.zip
         await Context.Tracing.StopAsync(new()
```
[the remark this answers is not in the document]
Comment (debs-obrien) [reply docs/src/trace-viewer-intro-csharp-java-python.md:176]: removed the comment as it is mentioned below. 
Need guidelines on how to add retain on failure

File: docs/src/writing-tests-csharp.md [hunk 3]
```diff
@@ -70,55 +92,109 @@ using Microsoft.VisualStudio.TestTools.UnitTesting;
...
+| [`method: Locator.press`] | Press single key |
+| [`method: Locator.setInputFiles`] | Pick files to upload |
+| [`method: Locator.selectOption`] | Select option in the drop down |
+
+## Assertions
+  
+Playwright provides an async function called [Expect](./test-assertions) to assert and wait until the expected condition is met.
+
+```csharp
+await Expect(Page).ToHaveTitleAsync(new Regex("Playwright"));
```
[the remark this answers is not in the document]
Comment (debs-obrien) [reply docs/src/writing-tests-csharp.md:178]: It is, but it aligns with JS and Python so if we change it we should change it everywhere. should we?

[the remark this answers is not in the document]
Comment (mxschmitt) [reply docs/src/writing-tests-csharp.md:178]: I think we should, lets follow-up with that.

File: docs/src/trace-viewer-intro-csharp-java-python.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (mxschmitt) [reply docs/src/trace-viewer-intro-csharp-java-python.md:172]: We can also follow-up with that.

**Review by mxschmitt — APPROVED** [review]

**Pull request merged**